book/style_guide.md

```diff
@@ -0,0 +1,221 @@
+                                             # two many spaces before "| $it + $acc": 1 instead of 0
+{x: 1, y : 2 } # two few spaces before "x: 1": 0 instead of 1
+               # two many spaces before ": 2": 1 instead of 0
+[1 2] | zip  [3 4] # two many spaces before "[3 4]": 2 instead of 1
+[ ] # two many spaces before "]": 1 instead of 0
+```
+
+#### Multi-line format
+
+Multi-line format is a format for writing all commands in several lines. It inherits all rules from one-line format
+and modifies them slightly.
+
+1. general:
+   1. **It's recommended to omit** trailing spaces.
+2. block and closure bodies:
+   1. **It's recommended to** put each body pipeline on a separate line.
+3. records:
+   1. **It's recommended to** put each record key-value pair on separate line.
+4. lists:
+   1. **It's recommended to** put each list item on separate line.
+5. surrounding constructs:
+   1. **It's recommended to** put one `\n` before opening square `[` or curly brace `{` if preceding symbol is not the same.
+   2. **It's recommended to** put one `\n` after closing square `]` or curly brace `}` if following symbol is not the same.
+
+Correct:
+
+```nushell
+[[status]; [UP] [UP]] | all {|el|
+    $el.status == UP
+}
+
+[
+    1
+    2
+    3
+    4
+] | reduce {|it, acc|
+    $it + $acc
+}
+
+{
+    x: 1,
+    y: 2
+}
+
+[
+  {
+    name: "Teresa",
+    age: 24
+  },
+  {
+    name: "Thomas",
+    age: 26
+  }
+]
+```
+
+Incorrect:
+
+```nushell
+[[status]; [UP] [UP]] | all {|el|
+    $el.status == UP } # two many spaces before "}": 1 instead of "\n"
+
+[ 1 # two many spaces before "1": 1 instead of "\n"
+    2
+    3
+    4
+] | reduce {|it, acc|
+    $it + $acc
+}
+
+{ x: 1, # two many spaces before "x: 1": 1 instead of "\n"
+    y: 2
+}
+
+[{ # two few "\n" before "{": 0 instead of 1
+    name: "Teresa",
+    age: 24
+  } , # two many spaces before ",": 1 instead of 0
+  {
+    name: "Thomas",
+    age: 26
+  }
+]
+```
+
+### Spreading long lines
+
+- **It's recommended to** default to short format unless you are writing scripts.
+- **It's recommended to** default to short format in scripts for lists and records unless they more than 80 characters long
+  or contain nested lists or records.
+- **It's recommended to** use long format for pipelines more than 80 characters long.
+
+Correct (in scripts):
+
+```nushell
+[1, 2, 3, 4] | reduce {|it, acc|
+    $it + $acc
+}
+
+[1 2 3 4] | reduce {|it acc|
+    $it + $acc
+}
+
+[
+  { name: "Teresa", age: 24 },
+  { name: "Thomas", age: 26 }
+]
+```
+
+Incorrect (in scripts):
+
+```nushell
+[
+   1,
+   2,
+   3,
+   4
+] | reduce {|it, acc|
+    $it + $acc
+}
+
+[
+   1
+   2
+   3
+   4
+] | reduce {|it acc|
+    $it + $acc
+}
+
+[
+  {
+    name: "Teresa",
+    age: 24
+  },
+  {
+    name: "Thomas",
+    age: 26
+  }
+]
+```
+
+## Options and parameters of custom commands
+
+- **It's recommended to** keep count of all positional parameters less than or equal to 4, for remaining inputs use options.
```

Comment:
i'd even bring that down to 1 unless a very specific command with extra-clear argument requirements :thinking: 
it very quickly becomes unreadable with strictly more that 1 positional :face_exhaling:

Reply:
I kind of agree with the idea with small amount of positional arguments. What about to narrow down to 2 instead of 4 assuming that some commands accepts _input_ and _output_ argument like `mv`?

Reply:
> - **It's recommended to** keep count of all positional parameters less than or equal to 2, for remaining inputs use options. Assume that command can expect source and destination parameter, like `mv`: source and target file or directory.

I've changed this point to this.

Reply:
yeah, 2 for commands with source and destination looks a good compromise :+1: 

other than that, i hardly see a use case where having more than 1 positional is a good and readable idea :open_mouth: